### ISSUE TITLE
Delete objects implementation is very sub-optimal

### ISSUE THREAD

**Issue by harshavardhana** [minio/console#1493]
https://github.com/minio/console/blob/master/restapi/user_objects.go#L621

```go
// deleteObjects deletes either a single object or multiple objects based on recursive flag
func deleteObjects(ctx context.Context, client MCClient, client2 MinioClient, bucket string, path string, versionID string, recursive bool, allVersions bool) error {
	if allVersions {
		if recursive {
			if err := deleteMultipleObjects(ctx, client, recursive, true); err != nil {
				return err
			}
		} else {
			objects, err := listBucketObjects(ctx, client2, bucket, path, recursive, true, false)
			if err != nil {
				return err
			}
			for i := range objects {
				if err := deleteSingleObject(ctx, client, bucket, path, objects[i].VersionID); err != nil {
					return err
				}
			}
		}
	}
	if recursive {
		if err := deleteMultipleObjects(ctx, client, recursive, false); err != nil {
			return err
		}
	} else {
		if err := deleteSingleObject(ctx, client, bucket, path, versionID); err != nil {
			return err
		}
	}
	return nil
}
```

Here many problems exist first 

- listBucketObjects() makes 4 extra calls for legal hold., retention, rags - these are useful for display but these are not useful to be re-queried to delete objects
- deleteSingleObject is called when deleting multiple objects
```go
			for i := range objects {
				if err := deleteSingleObject(ctx, client, bucket, path, objects[i].VersionID); err != nil {
					return err
				}
			}
```

- deleteSingleObject still invokes MultiObject delete API call (this is okay but the code should be combined here to call a single function that feeds objects to a common function through a contentCh

**Comment by harshavardhana** [comment]
@dvaldivia I think we need to increase priority on this issue.

**Comment by harshavardhana** [comment]
this has been fixed in #2153
